### ISSUE TITLE
Implement OptionalCollector

### ISSUE THREAD

**Issue by amaembo** [amaembo/streamex#87]
OptionalCollector is the collector specialization which return type is `Optional<R>` and it has methods which mimic the optional behavior (like `orElse`, `orElseGet`, `get`, `map`, `filter`, etc.). Should support both short-circuiting and non-short-circuiting collection.

**Comment by ThomasBlt** [comment]
@amaembo Isn't that solved by [MoreCollectors.onlyOne](https://github.com/amaembo/streamex/blob/d60b3497813958878f2a521747405c2f6b2a2536/src/main/java/one/util/streamex/MoreCollectors.java#L508) / findFirst() / findAny() ?

**Comment by amaembo** [comment]
No, the idea was to create a separate public type `OptionalCollector`, so you can do something like `MoreCollectors.onlyOne().orElse("foo")` as a shortcut to `Collectors.collectingAndThen(MoreCollectors.onlyOne(), opt -> opt.orElse("foo"))`.
